### PR TITLE
[MIN-92] QoL and bug fixes running end to end code

### PR DESCRIPTION
**Pull request by dudeperf3ct**
In this PR, we address few bugs and enhancement occurred when running all the pipeline end to end using instructions in readme. Following changes were made

- [x] DVC tags both data_scraping and data_preparation pipeline with same name
- [x] Change in data_embedding pipeline code data_version=current_data_version instead of data_version="data/first_version"  for embed step
- [x] Missing `reference_data_version` parameter in data_embedding config
- [x] Disable caching for version step used in both data_scraping and data_embedding pipeline
- [x] Readme with order of monitoring on k8s before running embedding pipeline
- [x] Add fixed parameters max_length and temperature while sending request to seldon endpoint.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +87,54 @@ Port-forward the chroma server service to localhost using the following command.
...
+
+Line number 39 in [monitoring-deployment.yaml](./infrastructure/monitoring/monitoring-deployment.yaml#L39) should be updated to match the Docker image name which we've just pushed to the ACR, and it will need to be in the following format: `<name-of-acr-registry>.azurecr.io/monitoring`.
+
+Next, we apply the Kubernetes manifest to deploy the metric service and the metric database on AKS.
+
+```bash
+kubectl apply -f infrastructure/monitoring
+```
+
+Finally, once the pod is running, we verify that our monitoring service is working. The command below should provide an IP address for the metric service interface.
```
Comment (JonoCX) [inline README.md]: Once `kubectl` has finished applying the manifest, we should verify that the monitoring service is running. Running the commands below will give you an IP address for the service, which we can then `curl` for a response:

File: README.md [hunk 2]
```diff
@@ -85,6 +87,54 @@ Port-forward the chroma server service to localhost using the following command.
...
+
+```bash
+kubectl get pods # Checking whether the monitoring pod is running
```
Comment (JonoCX) [inline README.md:118]: NIT: a snippet of output here would be useful, just the monitoring pod

File: README.md [hunk 3]
```diff
@@ -85,6 +87,54 @@ Port-forward the chroma server service to localhost using the following command.
...
+
+kubectl get svc monitoring-service -o jsonpath='{.status.loadBalancer.ingress[0].ip}'
+```
+
+We should be able to curl the external IP returned running the above command at port 5000.
```
Comment (JonoCX) [inline README.md:129]: If all the above works and shows that the service is running, we can use the external IP to `curl` the service on port 5000:

```bash
curl <external_ip>:5000
```

Question: this doesn't need a http or anything?

File: README.md [hunk 4]
```diff
@@ -85,6 +87,54 @@ Port-forward the chroma server service to localhost using the following command.
...
+
+```bash
+curl {external-ip:5000}
+
+# The response should be:
+Hello world from the metric service.
+```
+
+To compute the embedding drift when running the embedding pipeline, we will port-forward the monitoring service to localhost using the following command. This will ensure we can access the server from localhost.
```
Comment (JonoCX) [inline README.md]: Now we know that everything is up and running, we need to use port-forwarding so the embedding pipeline (which is running locally) can communicate with the service hosted on Kubernetes:

File: README.md [hunk 5]
```diff
@@ -178,54 +228,6 @@ or
 curl localhost:5000//query_embedding_drift
```
Comment (JonoCX) [inline README.md]: Double `//`?

**Review by JonoCX — CHANGES_REQUESTED** [review]
LGTM! A few suggested changes for the README to increase readability 🦖

File: README.md (hunk 5 above)
Comment (dudeperf3ct) [reply README.md]: Paging @osw282  for more insights on this.

Comment (osw282) [reply README.md]: Both single or double will work.

File: README.md (hunk 3 above)
Comment (dudeperf3ct) [reply README.md:129]: not required.

File: README.md (hunk 5 above)
Comment (JonoCX) [reply README.md]: Go with a single, it looks a bit bizarre

**Review by JonoCX — APPROVED** [review]
Nice one ✅

**Review by JonoCX — APPROVED** [review]

File: README.md
Comment (osw282) [inline README.md:1]: Just to confirm that this is the intended , there are two sections on monitoring one within the data scraping pipeline (run on k8s) and another one just on itself (run locally)?

**Review by osw282 — COMMENTED** [review]
One question on the README, looks good otherwise.

File: README.md
Comment (dudeperf3ct) [reply README.md:1]: One in data embedding pipeline (run on k8s). This section is required before running the data embedding pipeline. We require a monitoring service to be port-forwarded to run the pipeline.

Previously, monitoring was a separate section and the embedding pipeline would have failed if someone followed the instructions on Readme as is.  With the changes in place, before running the embedding pipeline we make sure both chroma and monitoring services are running and they are being port-forwarded as well.

File: pipelines/data_embedding_pipeline/config_data_embedding_pipeline.yaml [hunk 6]
```diff
@@ -2,6 +2,6 @@ steps:
   load_data:
     enable_cache: False
     parameters:
-      data_version: "data/first_version"
-      reference_data_version: "data/first_version"
+      data_version: "data/second_version"
```
Comment (d-lowl) [inline pipelines/data_embedding_pipeline/config_data_embedding_pipeline.yaml:5]: Why the change of version?

Comment (dudeperf3ct) [reply pipelines/data_embedding_pipeline/config_data_embedding_pipeline.yaml:5]: We have performed data cleaning using our experiment giving us a new version of data. For this new dataset, we are creating a new tagged dvc version.

**Review by osw282 — APPROVED** [review]
LGTM, well done Shubham! 🔥
